packages/atomic/src/components/result-template-components/atomic-result-quickview/atomic-result-quickview.tsx

```diff
@@ -0,0 +1,87 @@
+import {Component, State, h, Element} from '@stencil/core';
+import {
+  Result,
+  buildQuickview,
+  QuickviewState,
+  Quickview,
+} from '@coveo/headless';
+import {ResultContext} from '../result-template-decorators';
+import {
+  Bindings,
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+
+/**
+ * The ResultQuickview component renders a preview of the result.
+ */
+@Component({
+  tag: 'atomic-result-quickview',
+  styleUrl: 'atomic-result-quickview.pcss',
+  shadow: false,
+})
+export class AtomicResultQuickview implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @State() public error!: Error;
+
+  private quickview!: Quickview;
+
+  @BindStateToController('quickview')
+  @State()
+  private quickviewState!: QuickviewState;
+
+  @State() private isModalOpen = false;
+
+  @ResultContext() private result!: Result;
+
+  @Element() private host!: HTMLElement;
+
+  public initialize() {
+    const engine = this.bindings.engine;
+    const result = this.result;
+
+    this.quickview = buildQuickview(engine, {
+      options: {result},
+    });
+  }
+
+  private removeComponent() {
+    this.host.remove();
+  }
+
+  private openModal() {
+    this.quickview.fetchResultContent();
+    this.isModalOpen = true;
+  }
+
+  private closeModal() {
+    this.isModalOpen = false;
+  }
+
+  private get modal() {
+    return (
+      <atomic-modal handleClose={() => this.closeModal()}>
+        <iframe
```

Comment:
For the jsui we have these attributes:
```
sandbox: 'allow-popups allow-same-origin allow-top-navigation',
src: 'about:blank'
```

Reply:
I added the sandbox attribute, it's a good suggestion. I left out the `allow` values because I don't see why these exemptions are needed (maybe it's to do with the postmessage navigation?). We can always make things less strict when needed.

For the `src`, it seems this is an [outdated pattern](https://stackoverflow.com/questions/5946607/is-an-empty-iframe-src-valid/5946631#:~:text=You%20should%20use%20%3Ciframe%20src,when%20working%20with%20AJAX%20requests) for when browsers did not support iframes with no src attribute. We are also setting the `srcDoc` attribute in this case, the iframe is not blank.